go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -150,6 +152,19 @@ func buildDDLPlans(ctx context.Context, sql string, ddlStatement sqlparser.DDLSt
 		}, nil
 }
 
+func buildCreateProcedurePlan(vschema plancontext.VSchema, cp *sqlparser.CreateProcedure) (key.ShardDestination, *vindexes.Keyspace, error) {
+	destination, keyspace, _, err := vschema.TargetDestination(cp.Name.Qualifier.String())
+	if err != nil {
+		return nil, nil, err
+	}
+	// Clear out the qualifier from the table name.
+	cp.SetTable("", cp.Name.Name.String())
+	if keyspace.Sharded {
+		return nil, nil, vterrors.VT12001("CREATE PROCEDURE is not supported on sharded keyspaces")
+	}
```

Comment:
Two things here
1. We can check for sharded keyspace before replacing the qualifier
2. We need to replace the qualifier from the table name as well. We have methods in sqlparser that does that.

Reply:
Done